python/ambassador/ir/irhost.py

```diff
@@ -197,9 +207,13 @@ def setup(self, ir: "IR", aconf: Config) -> bool:
                             namespace=self.namespace,
                             location=self.location,
                             hosts=[self.hostname or self.name],
-                            secret=tls_name,
+                            secret=tls_full_name,
                         )
 
+                        # Ensure that we handle the secret properly even if tls_secret_namespacing is False in the Module
+                        if tls_namespace:
+                            tls_context_init["secret_namespacing"] = True
```

Comment:
Am I understanding this comment correctly...If I was a Platform Admin and set up a Module to disable this behavior. Then me as a developer comes along and says I don't care ignore it because I put? 

Or does this flag mean something else?

Reply:
The `Host` generates its own internal `IRTLSContext`. The `IRTLSContext` uses the questionable representation of `secret={name}, secret_namespacing=false` for non-cross-namespace secrets, and `secret={name}.{namespace}, secret_namespacing=true` for cross-namespace secrets.

But, you raise a good point that setting `secret_namespacing=False` in the Module effectively turns off the ability to do cross-namespace secrets in a manor similar to my proposed env-var, so perhaps we should disallow cross-namespace secrets if `ir.module.secret_namespacing==False`.

Reply:
In a TLSContext you can only specify secrets by providing a string such as `secret-name` or `secret-name.foo` By default secret namespacing is turned on. That tells a TLSContext that hey `secret-name.foo` is `secret-name` in the `foo` namespace. You can turn that behaviour off if you want, commonly because you might have a secret called `example.domain.net` where none of the `.`s are meant to indicate the namespace of the secret. 

When we create the implicit TLSContext for this Host we obviously want our TLSContext to treat the name as a namespace because one was explicitly supplied so I told the TLSContext to force the behaviour on otherwise if you have a Module that sets `tls_secret_namespacing: False` this feature won't work at all.

I think your comment about What if a platform Admin wants to disable the feature for security, not just to make sure that the format of the secrets is handled correctly has a point so I'll see if I can rework it real quick to not be dependent on how TLSContexts handle the secret name.

Reply:
Oh, but at the `ambassador` `Module` level it's just a default; so you can still override it on a per-`TLSContext` basis. So that's not any change here.

Reply:
Ok, thanks for details. That makes a ton more sense :)